### ISSUE TITLE
WordPress v4.4 Compat

### ISSUE THREAD

**Issue by jaswsinc** [websharks/comment-mail#170]
## Overview

WordPress v4.4 introduces some new objects for `WP_Term` and `WP_Comment`. This is mostly OK with Comment Mail. However, it recently came to my attention that in PHP 7.0 there are some flags being thrown with respect to type hints that are looking for `\stdClass` instances instead of the new `WP_Comment` class.

## What needs to change?

We need to search the codebase for `\stdClass $comment` and remove `\stdClass`.

**Comment by raamdev** [comment]
Closed by https://github.com/websharks/comment-mail/issues/171 and https://github.com/websharks/comment-mail-pro/issues/34

**Comment by raamdev** [comment]
Next Release Changelog:

- **Compatibility:** Updated for compatibility with WordPress v4.4 and PHP v7.0. Props @jaswsinc. See [Issue #170](https://github.com/websharks/comment-mail/issues/170).

**Comment by jaswsinc** [comment]
@raamdev I'm noting that this issue seems to impact all versions of PHP. Not just PHP 7+.

**Comment by raamdev** [comment]
> Not just PHP 7+

So this is entirely a WordPress v4.4 compatibility issue, i.e., Comment Mail is currently broken on WP 4.4 + any version of PHP?

**Comment by jaswsinc** [comment]
Correct. However, it's not something that totally breaks the site, or that you're going to notice as soon as you upgrade. From what I've seen in the lite version, it only occurs whenever you change the status of a comment.

```
Catchable fatal error: Argument 3 passed to comment_mail\plugin::comment_status() must be an instance of stdClass, instance of WP_Comment given in /home4/wsksdev/public_html/jason/wp-content/plugins/comment-mail/plugin.inc.php on line 2218
```

**Comment by raamdev** [comment]
Updated changelog entry to indicate that this was a bug fix for WP v4.4:

- **Bug Fix:** Fixed a compatibility bug with WordPress v4.4. Props @jaswsinc. See [Issue #170](https://github.com/websharks/comment-mail/issues/170).

**Comment by raamdev** [comment]
**Comment Mail v151224** has been released and includes changes from this GitHub Issue: See the [v151224 announcement](http://comment-mail.com/comment-mail-v151224-now-available/) for further details.

---

This issue will now be locked to further updates. If you have something to add related to this GitHub Issue, please open a new GitHub Issue and reference this one (#170).
